elastic/kibana
# [Files] Allow option to disable delete action in mgt UI

This PR adds the possibility to disable behaviour of files in the Files management UI. It is now possible to disable

* `list` - Files won't show in the management UI
* `delete` - It won't be possible to delete the file from the management UI

## Option added to the public registry: `managementUiActions`

```ts
files.registerFileKind({
      id: "...",
      allowedMimeTypes: "...",
      managementUiActions: { // NEW
        list: {
          enabled: false,
        },
        // If "list" is not enabled, this is not necessary :)
        // I leave it for documentation purpose
        delete: {
          enabled: false,
          reason: "Explain why the file cannot be deleted", // optional but nice for our users
        }
      },
    });
```

## How to test

* run kibana with the examples `yarn start --run-examples`
* open the "Files" example an upload a file in each table (all UI actions, no listing and no delete action)
* Navigate to the Management > Kibana > Files page and verify that the corresponding files are not shown or can't be deleted

## Screenshots

<img width="1605" alt="Screenshot 2023-04-18 at 16 41 25" src="https://user-images.githubusercontent.com/2854616/232830928-299fc3e5-b231-4935-90ec-97b50478c339.png">

<img width="1587" alt="Screenshot 2023-04-20 at 13 36 29" src="https://user-images.githubusercontent.com/2854616/233367861-b9b728c8-72cc-4c10-9e87-7063150fb19b.png">

Fixes https://github.com/elastic/kibana/issues/153756
